content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java

```diff
@@ -165,6 +166,77 @@ public void init() {
 		// Populate content upload headers used in uploadExternalContent
 		CONTENT_UPLOAD_HEADERS.putAll(BASE_HEADERS);
 		CONTENT_UPLOAD_HEADERS.put(HEADER_CONTENT, CONTENT_TYPE_BINARY);
+
+		try {
+			String webhookUrl = getWebhookUrl(Optional.empty());
+			if (!webhookUrl.contains("localhost")) {
+				webhookId = setupWebhook(webhookUrl);
+				getWebhook(webhookId);
+			} else {
+				log.info("Running locally - ignoring webhook calls");
+			}
+		} catch (Exception e) {
+			log.error(e.getLocalizedMessage());
+		}
+	}
+
+	public String setupWebhook(String webhookUrl) throws IOException {
+		String id = "";
+
+		Map<String, Object> data = new HashMap<String, Object>();
+
+		List<String> types = new ArrayList<>();
+		types.add("SIMILARITY_COMPLETE");
+		types.add("SUBMISSION_COMPLETE");
+
+		data.put("signing_secret", apiKey);
+		data.put("url", webhookUrl);
+		data.put("description", "test");
+		data.put("allow-insecure", false);
+		data.put("event-types", types);
```

Comment:
you could use a shorthand: Arrays.asList("SIMILARITY_COMPLETE", "SUBMISSION_COMPLETE") and get rid of the "types" object